lib/internal/http2/core.js

```diff
@@ -717,6 +719,17 @@ function setupHandle(socket, type, options) {
 
   this[kHandle] = handle;
 
+  if (socket.encrypted) {
+    this[kAlpnProtocol] = socket.alpnProtocol;
+    this[kEncrypted] = true;
+    const origin =
+      (new URL(`https://${socket.servername}:${socket.remotePort}`)).origin;
+    this[kState].originSet = new Set([origin]);
```

Comment:
Can you please add a TODO here that the `Set` will be used by the ORIGIN frame support?

Reply:
Will do